src/RawPOSIX/src/safeposix/dispatcher.rs

```diff
@@ -775,11 +776,52 @@ pub fn lind_syscall_api(
         SHMAT_SYSCALL => {
             let shmid = arg1 as i32;
             let cage = interface::cagetable_getref(cageid);
-            // Convert virtual address to physical address
             let shmaddr = translate_vmmap_addr(&cage, arg2).unwrap() as *mut u8;
             let shmflg = arg3 as i32;
-            // Perform shmat operation through cage implementation
-            cage.shmat_syscall(shmid, shmaddr, shmflg)
+            
+            // Get reference to global shared memory metadata
```

Comment:
I believe that the code before the handler call as well as the code after can be moved into the shmat_handler or into the syscall. We should probably do that to match how the dispatcher works elsewhere (ie not doing critical functionality in the dispatcher)